src/dfm/data/load_datasets.py

```diff
@@ -0,0 +1,217 @@
+
+from typing import Union, Iterable, List, Optional
+import os
+from pathlib import Path
+from functools import partial
+
+from datasets import (
+    load_dataset,
+    interleave_datasets,
+    DatasetDict,
+    IterableDatasetDict,
+    load_from_disk,
+    interleave_datasets,
+    concatenate_datasets
+)
+
+HOPETWITTER_PATH = Path("/work") / "twitter_cleaned"
+DAGW_DFM_PATH = Path("/work") / "dagw-clean"
+DANEWS_PATH = Path("/work") / "hope-infomedia_cleaned"
+NAT_PATH = Path("/work") / "netarkivet-cleaned"
+
+
+
+def __add_source(example, source):
+    example["source"] = source
+    return example
+
+def keep_columns(
+    dataset: Union[IterableDatasetDict, DatasetDict], 
+    columns_to_keep: List[str]
+) -> Union[IterableDatasetDict, DatasetDict]:
+    subset = dataset[list(dataset.keys())[0]]
+    sample = next(iter(subset))
+    col_to_remove = [c_name for c_name in sample.keys() if c_name not in columns_to_keep]
+    
+    return dataset.remove_columns(col_to_remove)
+
+
+
+def load_hopetwitter(
+        path_to_hopetwitter: Union[str, Path]=HOPETWITTER_PATH, 
+        columns_to_keep: Optional[List[str]]=None, 
+        repeats: int=1,
+        **kwargs
+    ) -> IterableDatasetDict:
+    """
+    Loads HopeTwitter.
+
+    Args:
+        path_to_hopetwitter (Union[str, Path, None], optional): path to the HopeTwitter.
+            Defaults to "/work/twitter_cleaned".
+        columns_to_keep (Optional[List[str]], optional): Columns to keep. Default to None
```

Comment:
```suggestion
        columns_to_keep (Optional[List[str]], optional): Columns to keep. Default to None
```

Redundant optional? ;-)

Reply:
The first optional is the type hint, indicating that it can also be None, the second optional indicated the variable is optional (it is not needed to specify it).